### ISSUE TITLE
[Bug]: Triggers on ECS Service don't result in changes

### ISSUE THREAD

**Issue by creeder-uturn** [hashicorp/terraform-provider-aws#29241]
### Terraform Core Version

1.2.9, 1.3.7

### AWS Provider Version

4.40.0, 4.53.0

### Affected Resource(s)

aws_ecs_service

### Expected Behavior

When using the triggers = {} parameter (introduced here: https://github.com/hashicorp/terraform-provider-aws/pull/25840), changes to the triggers should result in a change in the terraform resource. 

### Actual Behavior

Changing the triggers = {}, including removal, results in no difference in plan

### Terraform Configuration Files

terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 4.40"
    }
  }
}

resource "aws_ecs_cluster" "main" {
  name = "bug-demo"
}

resource "aws_ecs_service" "main" {
  name            = "bug-demo"
  cluster         = aws_ecs_cluster.main.id
  task_definition = aws_ecs_task_definition.main.arn

  force_new_deployment = true

  # Triggers a change in the service resource even if nothing changes, when force_new_deployment is true
  triggers = {
    redeployment = timestamp()
  }
}

resource "aws_ecs_task_definition" "main" {
  family = "bug-demo"

  container_definitions = jsonencode([{
    name      = "sleep"
    image     = "busybox"
    cpu       = 10
    memory    = 10
    command   = ["sleep", "360"]
    essential = true
  }])
}

### Steps to Reproduce

Apply the above terraform. Apply again. timestamp() should result in triggers changing. It does not.

### References

This seems to be when this feature was introduced, and as far as I can tell it has never worked: https://github.com/hashicorp/terraform-provider-aws/pull/25840

### Would you like to implement a fix?

No

**Comment by creeder-uturn** [comment]
Additional finding around this feature: if you flip `force_new_deployment` to false in the above test, apply, then flip it back to true, you get:

```
│ Error: Provider produced inconsistent final plan
│
│ When expanding the plan for aws_ecs_service.main to include new values learned so far during apply, provider "registry.terraform.io/hashicorp/aws" produced an invalid new
│ value for .triggers["redeployment"]: was cty.StringVal("2023-02-03T17:53:36Z"), but now cty.StringVal("2023-02-03T18:08:09Z").
│
│ This is a bug in the provider, which should be reported in the provider's own issue tracker.
```

**Comment by gtskaushik** [comment]
I'm facing the same issue. A change in triggers is not detected during plan or apply. Also when trying apply multiple times, I ended up in this error

> When expanding the plan for aws_ecs_service.main to include new values learned so far during apply, provider "registry.terraform.io/hashicorp/aws" produced an invalid new
│ value for .triggers["redeployment"]: was cty.StringVal("2023-02-03T17:53:36Z"), but now cty.StringVal("2023-02-03T18:08:09Z").

**Comment by gtskaushik** [comment]
This looks like a duplicate of https://github.com/hashicorp/terraform-provider-aws/issues/28070

**Comment by gtskaushik** [comment]
I found a workaround for this. 
1. Create a new variable
`variable "timestamp_id" {
    type = string
    default = ""
}`
2. Pass the value during plan or apply
`-var="timestamp_id=$(date)`
3. Change in aws_ecs_service
`
force_new_deployment = true
  triggers = {
    redeployment = var.timestamp_id
  }
`

**Comment by jonathan-heartland** [comment]
I am encountering this same problem. 

`Error: Provider produced inconsistent final plan
│ 
│ When expanding the plan for
│ module.ecs.aws_ecs_service.instance["main"] to include new values
│ learned so far during apply, provider "registry.terraform.io/hashicorp/aws"
│ produced an invalid new value for .triggers["redeployment"]: was
│ cty.StringVal("2023-05-02T16:13:05Z"), but now
│ cty.StringVal("2023-05-02T17:01:34Z").
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.`

My Terraform Apply's are all pipelined, so adding a workaround is not ideal.

**Comment by lordz-ei** [comment]
> I am encountering this same problem.
> 
> `Error: Provider produced inconsistent final plan │ │ When expanding the plan for │ module.ecs.aws_ecs_service.instance["main"] to include new values │ learned so far during apply, provider "registry.terraform.io/hashicorp/aws" │ produced an invalid new value for .triggers["redeployment"]: was │ cty.StringVal("2023-05-02T16:13:05Z"), but now │ cty.StringVal("2023-05-02T17:01:34Z"). │ │ This is a bug in the provider, which should be reported in the provider's │ own issue tracker.`
> 
> My Terraform Apply's are all pipelined, so adding a workaround is not ideal.

Same issue here

**Comment by Surgo** [comment]
It's works fine for me.

```terraform
  triggers = {
    redeployment = plantimestamp()
  }
```

**Comment by aszniak** [comment]
Right, it works for `plantimestamp()`, but I wanted to make it redeploy on config change. 
Something along the lines of:

```
 triggers = {
    redeployment = filesha256("${path.module}/files/service-config.yml")
 }
```

Does not seem to work

**Comment by ngyuki** [comment]
I came here looking for way to redeployment ecs service when secretsmanager is changed.

It does not seem to work if aws_ecs_service.triggers contains computed value.

```tf
resource "aws_ecs_service" "main" {

  # ...snip...

  force_new_deployment = true
  triggers = {
    # does not works
    secret_id  = aws_secretsmanager_secret_version.main.secret_id
    version_id = aws_secretsmanager_secret_version.main.version_id
  }
}
```

It works by inserting terraform_data as follows.

```tf
resource "aws_ecs_service" "main" {

  # ...snip...

  force_new_deployment = true
  triggers = {
    # it's works
    redeployment = terraform_data.triggers.input
  }
}

resource "terraform_data" "triggers" {
  input = plantimestamp()
  triggers_replace = {
    secret_id  = aws_secretsmanager_secret_version.main.secret_id
    version_id = aws_secretsmanager_secret_version.main.version_id
  }
  lifecycle {
    ignore_changes = [input]
  }
}
```
